common/desktop-exports

```diff
@@ -102,20 +104,62 @@ mkdir -p $XDG_CONFIG_HOME
 # Ensure the app finds locale definitions (requires locales-all to be installed)
 append_dir LOCPATH $RUNTIME/usr/lib/locale
 
-# Run xdg-user-dirs-update
-if [ `which xdg-user-dirs-update` ]; then
-   xdg-user-dirs-update
+# If any, keep track of where XDG dirs were so we can potentially migrate the content later
+test -f ${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs && . ${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs
+for d in DOCUMENTS DESKTOP DOWNLOAD MUSIC PICTURES VIDEOS PUBLICSHARE TEMPLATES; do
+  eval $(echo "OLD_XDG_${d}_DIR")=`eval "$(echo "echo \\$XDG_${d}_DIR")"`
+done
+
+# Setup user-dirs.* or run xdg-user-dirs-update if needed
+needs_xdg_update=false
+needs_xdg_links=false
+if can_open_file "$REALHOME/.config/user-dirs.dirs" && can_open_file "$REALHOME/.config/user-dirs.locale"; then
+   sed s#\$HOME#${REALHOME}#g $REALHOME/.config/user-dirs.dirs > $HOME/.config/user-dirs.dirs
```

Comment:
That's a harmless detail, but this will also replace the "$HOME" in the header of the file, which says:
    # Format is XDG_xxx_DIR="$HOME/yyy", where yyy is a shell-escaped
Ideally this would be replaced only in the 'XDG_xxx_DIR="$HOME/yyy"' lines.

Reply:
Yeah, agree... Using something like this (that will skips the lines) should be better: https://stackoverflow.com/a/4798161/210151

Reply:
Done!